### PR TITLE
Allow users to specify host alias in azkaban property file.

### PR DESCRIPTION
**Pull request by chengren311**
User can fill in their own alias in property file. It will be used as hostname in getHost(). If not configured we will use canonical hostname as fallback.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/constants/ServerInternals.java [hunk 1]
```diff
@@ -26,5 +26,9 @@
   public static final String AZKABAN_EXECUTOR_PORT_FILENAME = "executor.port";
 
   public static final String AZKABAN_SERVLET_CONTEXT_KEY = "azkaban_app";
-  
+
+  //property names
+  public static final String AZKABAN_HOST_ALIAS = "azkaban.hostalias";
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerInternals.java]: This config is exposed to the users. Shouldn't it be in azkaban.constants.ServerProperties and follow the naming convention there?

**Comment by chengren311** [comment]
@HappyRay updated the rb. just realized there's already "executor.host" property we can leverage here.

File: azkaban-common/src/main/java/azkaban/constants/ServerInternals.java [hunk 2]
```diff
@@ -26,5 +26,5 @@
   public static final String AZKABAN_EXECUTOR_PORT_FILENAME = "executor.port";
 
   public static final String AZKABAN_SERVLET_CONTEXT_KEY = "azkaban_app";
-  
-}
+
+}
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerInternals.java]: Is this change needed?

File: azkaban-common/src/main/java/azkaban/constants/ServerProperties.java [hunk 3]
```diff
@@ -47,4 +47,7 @@
 
   public static final String IS_METRICS_ENABLED =
       "azkaban.is.metrics.enabled";
+
+  public static final String EXECUTOR_HOST = "executor.host";
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: Could you add a comment here that this section of the constants are legacy configs?

The new configs should follow the naming convention of azkaban.server.<name> for server configs.

**Comment by kunkun-tang** [comment]
The PR let us have custom host name given one executor. I'm wondering how could we get port

File: azkaban-common/src/main/java/azkaban/constants/ServerProperties.java [hunk 4]
```diff
@@ -47,4 +47,12 @@
 
   public static final String IS_METRICS_ENABLED =
       "azkaban.is.metrics.enabled";
+
+  // Hostname for the host, if not specified, canonical hostname will be used
+  public static final String AZKABAN_SERVER_HOST_NAME = "azkaban.server.hostname";
+
+  // Legacy configs section, new configs should follow the naming convention of azkaban.server. for server configs
+
```
Comment (shuzhang1989) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: nit: empty line

**Comment by chengren311** [comment]
@kunkun-tang why do we need to know port here?

File: azkaban-common/src/main/java/azkaban/constants/ServerProperties.java [hunk 5]
```diff
@@ -48,6 +48,11 @@
   public static final String IS_METRICS_ENABLED =
       "azkaban.is.metrics.enabled";
 
-  public static final String EXECUTOR_HOST = "executor.host";
+  // Hostname for the host, if not specified, canonical hostname will be used
+  public static final String AZKABAN_SERVER_HOST_NAME = "azkaban.server.hostname";
+
+  // Legacy configs section, new configs should follow the naming convention of azkaban.server. for server configs
 
+  // The property is used for web server to get the host name of the executor when running in SOLO mode,
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: "the web server"?

Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: Consider making this a separate re-factor pull request? If it is too much trouble since you've made the change, could you at least add it to the change description?

File: azkaban-common/src/main/java/azkaban/constants/ServerInternals.java [hunk 6]
```diff
@@ -26,5 +26,4 @@
   public static final String AZKABAN_EXECUTOR_PORT_FILENAME = "executor.port";
 
   public static final String AZKABAN_SERVLET_CONTEXT_KEY = "azkaban_app";
-  
-}
+}
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerInternals.java]: Is this change still necessary?

File: azkaban-common/src/main/java/azkaban/constants/ServerProperties.java (hunk 4 above)
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: @shuzhang1989 thanks for reviewing the change! Do you mean removing this empty line?

The empty line helps to separate the heading of this section from the rest.

File: azkaban-common/src/main/java/azkaban/constants/ServerProperties.java (hunk 5 above)
Comment (chengren311) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: executor.host is used in executor manager only, which is inside web server.

File: azkaban-common/src/main/java/azkaban/constants/ServerProperties.java [hunk 7]
```diff
@@ -47,4 +47,12 @@
 
   public static final String IS_METRICS_ENABLED =
       "azkaban.is.metrics.enabled";
+
+  // Hostname for the host, if not specified, canonical hostname will be used
+  public static final String AZKABAN_SERVER_HOST_NAME = "azkaban.server.hostname";
+
+  // Legacy configs section, new configs should follow the naming convention of azkaban.server. for server configs
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: Consider azkaban.server.<rest of the name> to make it more explicit.
Add '.' to end a sentence.

File: azkaban-common/src/main/java/azkaban/constants/ServerProperties.java (hunk 5 above)
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: I meant the grammar in the comment.

**Comment by kunkun-tang** [comment]
@chengren311 ELK logs needs to differentiate executors by executor_id or port. Besides, I'm seeing that **_getHost_** is not a static member. How could other modules know existing host name, like ELK?

File: azkaban-common/src/main/java/azkaban/constants/ServerProperties.java (hunk 4 above)
Comment (shuzhang1989) [inline azkaban-common/src/main/java/azkaban/constants/ServerProperties.java]: i see, i thought `for server configs` was an unfinished sentence. thanks @HappyRay for the explanation :)

**Comment by chengren311** [comment]
@kunkun-tang we do have the getPort() method in the same class AzkabanExecutorServer. I think we will need to make it singleton so that we can use getHost(), getPort() by doing like AzkabanExecutorServer.getInstance().getHost(). We can do it in a separate commit.

**Comment by kunkun-tang** [comment]
I see. Thanks, @chengren311 . One final thing confused me is how to use. All executors share the same configuration, right? how could AZKABAN_SERVER_HOST_NAME apply for an specific executor?

**Comment by chengren311** [comment]
@kunkun-tang yes, we will backfill this property when deploying the config file to each host.

**Comment by HappyRay** [comment]
Please remember to update your change description before you merge. Please include a short description of your refactoring change.
